Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,426 @@
+*/
+void readInputs(){
+
+    #ifndef __TEST_SENDTRITIUM
+    Minion_Error_t err;
+
+    brakePedalPercent = Pedals_Read(BRAKE);
+    accelPedalPercent = Pedals_Read(ACCELERATOR);
+    
+    // UPDATE BUTTONS
+    if(Minion_Read_Input(REGEN_SW, &err) && regenCounter < DEBOUNCE_PERIOD){regenCounter++;}
+    else if(regenCounter > 0){regenCounter--;}
+
+    if(Minion_Read_Input(CRUZ_EN, &err) && cruiseEnableCounter < DEBOUNCE_PERIOD){cruiseEnableCounter++;}
+    else if(cruiseEnableCounter > 0){cruiseEnableCounter--;}
+
+    if(Minion_Read_Input(CRUZ_EN, &err) && cruiseSetCounter < DEBOUNCE_PERIOD){cruiseSetCounter++;}
+    else if(cruiseSetCounter > 0){cruiseSetCounter--;}
+    
+    forwardSwitch = Minion_Read_Input(FOR_SW, &err);
+    reverseSwitch = Minion_Read_Input(REV_SW, &err);
+
+    if(forwardSwitch){
+        cruiseSet = Minion_Read_Input(CRUZ_ST, &err);
+    }
+    else{
+        cruiseEnable = false;
+        cruiseSet = false;
+        regenToggle = false;
+    }
+
+
+    // DEBOUNCING
+    if(regenCounter == DEBOUNCE_PERIOD){regenButton = true;}
+    else if(regenCounter == 0){regenButton = false;}
+
+    if(cruiseEnableCounter == DEBOUNCE_PERIOD){cruiseEnableButton = true;}
+    else if(cruiseEnableCounter == 0){cruiseEnableButton = false;}
+
+    if(cruiseSetCounter == DEBOUNCE_PERIOD){cruiseSet = true;}
+    else if(cruiseSetCounter == 0){cruiseSet = false;}
+
+    // TOGGLE
+    if(ChargeEnable && regenButton != regenPrevious && regenPrevious){regenToggle = !regenToggle;}
+    regenPrevious = regenButton;
+    
+    if(cruiseEnableButton != cruiseEnablePrevious && cruiseEnablePrevious){cruiseEnable = !cruiseEnable;}
+    cruiseEnablePrevious = cruiseEnableButton;
+
+    #endif
+}
+
+/**
+ * @brief Converts integer percentage to float percentage
+ * @param percent integer percentage from 0-100
+*/
+extern const float pedalToPercent[];
+static float percentToFloat(uint8_t percent){
+    if(percent > 100){
+        return 1.0f;
+    }
+    return pedalToPercent[percent];
+}
+
+/**
+ * @brief Follows the FSM to update the velocity of the car
+*/
+void Task_SendTritium(void *p_arg){
+    OS_ERR err;
+    Minion_Error_t minion_err;
+
+#ifndef __TEST_SENDTRITIUM
+    CANPayload_t velocityPayload;
+            velocityPayload.bytes = 32;
+            velocityPayload.data.d = velocitySetpoint;
+
+    CANPayload_t currentPayload;
+            currentPayload.bytes = 32;
+            currentPayload.data.d = currentSetpoint;
+#endif
+
+    while(1){
+        FSM[state].stateHandler();    // do what the current state does
+        readInputs();   // read inputs from the system
+        FSM[state].stateDecider();    // decide what the next state is
+
+        // Sets brakelight
+        Minion_Write_Output(BRAKELIGHT, brakelightState, &minion_err);
+
+        // Drive
+        #ifndef __TEST_SENDTRITIUM
+         if(MOTOR_MSG_COUNTER_THRESHOLD%MotorCounter == 0){ // if smaller numbers, MOTOR_MSG_COUNTER_THRESHOLD-MotorCounter == 0
```

Comment:
If you're resetting the Motor Counter every loop, can't you just check if `MotorCounter == MOTOR_MSG_COUNTER_THRESHOLD`?